.github/workflows/rust.yml

```diff
@@ -94,31 +94,63 @@ jobs:
     strategy:
       matrix:
         source_os: [ubuntu-latest, windows-latest, local]
+    env:
+      api-level: 29
+
     steps:
     - name: Checkout
       uses: actions/checkout@v2
+
     - uses: actions/download-artifact@v2
       name: Download hello_world APK
       if: ${{ matrix.source_os != 'local' }}
       id: download
       with:
         name: hello_world_${{ matrix.source_os }}_x86_64-linux-android
+
     - name: Install `cargo-apk` and add `x86_64-linux-android` target
       if: ${{ matrix.source_os == 'local' }}
       run: |
         cargo install --path cargo-apk
         rustup target add x86_64-linux-android
-    - name: Start hello_world example
+
+    - name: AVD cache
+      uses: actions/cache@v2
+      id: avd-cache
+      with:
+        path: |
+          ~/.android/avd/*
+          ~/.android/adb*
+        # Bump the trailing number when making changes to the emulator setup below
+        key: avd-${{ env.api-level }}-1
+
+    - name: create AVD and generate snapshot for caching
+      if: steps.avd-cache.outputs.cache-hit != 'true'
       uses: reactivecircus/android-emulator-runner@v2
       with:
-        api-level: 29
+        api-level: ${{ env.api-level }}
         arch: x86_64
         # the `googleapis` emulator target is considerably slower on CI.
         target: default
         profile: Nexus 6
         emulator-options: -no-window -gpu swiftshader_indirect -noaudio -no-boot-anim -camera-back none
         disable-animations: true
+        force-avd-creation: false
```

Comment:
@Gordon-F the example in their readme also sets `force-avd-creation` to `false`. That seems like a sensible default so perhaps we shouldn't even set this at all. I don't mind it not creating an `avd` when it already exists, though that is extremely unlikely/impossible unless GH actions environment starts to ship with one...